fpga/pulpissimo-zedboard/Makefile

```diff
@@ -19,26 +18,29 @@ ips: clk ram ## Synthesizes necessary xilinx IP
 clean-ips: clean-clk clean-ram ## Clean all IPs
 
 clk: ## Synthesizes the Xilinx Clocking Manager IPs
-	cd ips/xilinx_clk_mngr; make all
-	cd ips/xilinx_slow_clk_mngr; make all
+	$(MAKE) -C ips/xilinx_clk_mngr all
+	$(MAKE) -C ips/xilinx_slow_clk_mngr all
 
 clean-clk: ## Removes all Clocking Wizard IP outputs
-	cd ips/xilinx_clk_mngr; make clean
-	cd ips/xilinx_slow_clk_mngr; make clean
+	$(MAKE) -C ips/xilinx_clk_mngr clean
+	$(MAKE) -C ips/xilinx_slow_clk_mngr clean
 
 ram: ## Synthesizes the Xilinx Block Memory Generator IPs for PULPissimo's L2 Ram
-	cd ips/xilinx_interleaved_ram; make all
-	cd ips/xilinx_private_ram; make all
+	$(MAKE) -C ips/xilinx_interleaved_ram all
+	$(MAKE) -C ips/xilinx_private_ram all
 
 clean-ram: ## Removes all Block Ram IP outputs related to l2 ram
-	cd ips/xilinx_interleaved_ram; make clean
-	cd ips/xilinx_private_ram; make clean
+	$(MAKE) -C ips/xilinx_interleaved_ram clean
+	$(MAKE) -C ips/xilinx_private_ram clean
 
 clean: ## Removes all bitstreams, *.log files and vivado related files (rm -rf vivado*)
 	rm -rf ${PROJECT}.*[^'bit']
 	rm -rf ${PROJECT}.*[^'bin']
 	rm -rf *.log
 	rm -rf vivado*
 
+download: ## Download the bitstream into the FPGA
+	$(VIVADO) -mode batch -source tcl/download_bitstream.tcl -tclargs $(PROJECT).runs/impl_1/xilinx_pulpissimo.bit pulpissimo_$(BOARD).bit
+
```

Comment:
I'm not entirely familiar with this: Does this path always exists `$(PROJECT).runs/impl_1/xilinx_pulpissimo.bit` ? Especially the `impl_1` looks a bit suspicious.

Reply:
That's the default name used by Vivado and referenced pretty much everywhere (makefile and tcl scripts), cf. `git grep impl_1`. The main makefile even copies those bitstream files using that path.